client/client.go

```
@@ -62,6 +63,11 @@ func processResponse(res *http.Response, decodeTo interface{}) error {
 }
 
 func (c *WaspClient) do(method, route string, reqObj, resObj interface{}) error {
+	if c.baseURL == "" {
+		log.Printf("wasp webapi not defined, request to %s not done", route)
+		return nil
```

Why not return an error here?

Actually shoudn't the `WaspClient` constructor panic instead if `baseURL` is empty?

---

I didn't wanna error because all over wasp cli this gets called, just wanted to print that no calls are being made because the user didn't specify a webapi url

---

But this doesn't belong exclusively to wasp-cli, it is the generic wasp client object. It could be used anywhere, so printing to the console is not correct :-\

---

Sorry for insisting, but why not panic? Maybe I'm missing something but calling a constructor that returns nil sounds like a smell :grimacing: .

---

>sounds like a smell
:smile: 

cli code calls: `client := config.WaspClient()` when trying to query an api. What should that return when there is no such api defined in `wasp-cli.json`? 

I guess the design flaw here is that this shouldn't receive an optional index and try to find the config after the fact, but rather just receive it as a parameter in the constructor:
https://github.com/iotaledger/wasp/blob/develop/tools/wasp-cli/config/config.go#L161

---

Yes, I see you already changed that :+1: 

So now this condition should never happen?

```
if c.baseURL == ""
```

It's safe to remove this check or panic?